webapp/src/client.js

```diff
@@ -1,20 +1,25 @@
 import {id as pluginId} from './manifest';
 
+const HEADER_X_CSRF_TOKEN = 'X-CSRF-Token';
+
 export class Client {
     constructor() {
         this.url = `/plugins/${pluginId}/api/v1`;
     }
 
     connected = () => {
-        return this.doFetch(`${this.url}/connected`, {
-            method: 'POST',
-            headers: {
-                'X-Requested-With': 'XMLHttpRequest',
-            },
-        });
+        return this.doFetch(`${this.url}/connected`, {method: 'POST'});
     }
 
     doFetch = async (url, options) => {
+        if (!options.headers) {
+            options.headers = {};
+        }
+
+        if (options.method && options.method.toLowerCase() !== 'get') {
+            options.headers[HEADER_X_CSRF_TOKEN] = getCSRFFromCookie();
```

Comment:
Would it make sense to cache the cookie in the Client constructor as a property?

Reply:
We could cache it, but I don't think passing it into the constructor is necessary. The CSRF token doesn't ever change, does it?

Reply:
Right - I've meant to get the cookie value in the constructor initially once and cache it in a Client class property. Sorry, that was badly worded.